packages/type-storage/src/types.d.ts

```diff
@@ -2,22 +2,21 @@
 // This software may be modified and distributed under the terms
 // of the ISC license. See the LICENSE file for details.
 
-import BN from 'bn.js';
-import { Section } from '@polkadot/params/types';
+import { IStorageFunctionMetadata } from '@polkadot/api-codec/Metadata';
 
-export type Storage$Key$Value = number | BN | Uint8Array | string;
+import * as wellKnownKeys from './wellKnownKeys';
 
-export type Storages = {
-  consensus: Section<Storages, any, any>,
-  // contract: Section<Storages, any, any>,
-  council: Section<Storages, any, any>,
-  councilVoting: Section<Storages, any, any>,
-  democracy: Section<Storages, any, any>,
-  parachains: Section<Storages, any, any>,
-  session: Section<Storages, any, any>,
-  staking: Section<Storages, any, any>,
-  system: Section<Storages, any, any>,
-  timestamp: Section<Storages, any, any>
-};
+type WellKnownKeys = keyof typeof wellKnownKeys;
 
-export type Storage$Sections = keyof Storages;
+export interface StorageFunction extends IStorageFunctionMetadata {
```

Comment:
I'm guessing this could also be (so we don't define the `IStorageFunctionMetadata`, something like -

```
import { StorageFunctionMetadata } from '@polkadot/api-codec/Metadata;

export interface StorageFunction {
  (...args: any[]): any;
  [K in keyof StorageFunctionMetadata]: StorageFunctionMetadata[K];
}
```

(Nothing wrong here, just giving options in case we don't want to define the same thing twice, the class basically has all the values anyway. Really dunno what is best-practice, but I know that I've had interface fatigue a lot in these...)